lib/fragment.js

```diff
@@ -197,17 +198,18 @@ module.exports = class Fragment extends EventEmitter {
      * Insert the placeholder for pipe assets at the end of fragment stream
      */
     insertEnd () {
-        this.index--;
         if (this.scriptRefs.length > 0) {
+            const range = [this.index - this.scriptRefs.length, this.index - 1];
+            this.index--;
+            const fragmentId = this.attributes.id || range[0];
```

Comment:
By keeping the id consistent, It ll become easy to measure the fragment perf. I will update the fragment-performance script to handle this case.